### ISSUE TITLE
“error 25: Missing field 'authors'” or “error 23: Missing field 'maintainer'” when trying 2.0 beta

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [ocaml/opam#3192]
I'm a newbie, but I wanted to install and get used to the upcoming forms of all the OCaml-ecosystem tooling.

However in my toy project, having followed [the documentation](https://opam.ocaml.org/doc/Packaging.html) for creating a package,

```
opam-version:  "1.2"

name:          "ocameel"
version:       "0.0.1"
maintainer:    "ELLIOTTCABLE <me@ell.io>"
license:       "ISC"

homepage:      "https://github.com/ELLIOTTCABLE/ocameel"
bug-reports:   "https://github.com/ELLIOTTCABLE/ocameel/issues"
dev-repo:      "https://github.com/ELLIOTTCABLE/ocameel.git"

build: [
  ["jbuilder" "subst" "-n" name] {pinned}
  ["jbuilder" "build" "-p" name "-j" jobs]
]
build-test: ["jbuilder" "build" "test/git"]

depends: [
  "jbuilder" {build}
]
```

Note, particularly, the line “The `opam-version`, `maintainer` and `version` fields are mandatory ; you should remove the others rather than leave them empty.”

Well, after installing v2.0-beta5, I get this after creating a new, local, switch:

```
$ ocameel master* opam pin add ocameel . -n
[NOTE] Package ocameel is already pinned to git+file:///Users/ec/Dropbox/Code/ocameel#master
       (version 0.0.1).
[ocameel.0.0.1] no changes from git+file:///Users/ec/Dropbox/Code/ocameel#master
[NOTE] Converting format of
       /Users/ec/Dropbox/Code/ocameel/_opam/.opam-switch/sources/ocameel/ocameel.opam from 1.2 to
       2.0~beta5
[WARNING] Failed checks on ocameel package definition from source at
          git+file:///Users/ec/Dropbox/Code/ocameel#master:
    error 25: Missing field 'authors'
ocameel is now pinned to git+file:///Users/ec/Dropbox/Code/ocameel#master (version 0.0.1)
```

It seems that documentation is out-of-date; so I think v2.0 should probably gracefully handle a lack of `author`, `authors`, or `maintainers`, when updating the `.opam` file to the v2.0 format!

Just a thought! `<3`

**Comment by ELLIOTTCABLE** [comment]
On a related note, but not really requiring a separate Issue — is there any tooling or features for *maintaining* the `.opam` file? Do I need to edit this manually, while referring to the syntax documentation; or is there something along the lines of `npm install --save`, or `npm init`, to get `opam` to manage it for me?

**Comment by ELLIOTTCABLE** [comment]
Additionally, the v2.0 manual [*still* only lists `maintainers`](https://opam.ocaml.org/doc/2.0/Manual.html#opam-root) as the only mandatory field, and yet `opam` commands complain after having everything else updated to the v2.0 format:

```
$ ocameel master* [6s] opam pin add ocameel . -n -k path
[NOTE] Package ocameel is already pinned to file:///Users/ec/Dropbox/Code/ocameel (version
       0.0.1).
[ocameel.0.0.1] synchronised from file:///Users/ec/Dropbox/Code/ocameel
[WARNING] Failed checks on ocameel package definition from source at
          file:///Users/ec/Dropbox/Code/ocameel:
    error 25: Missing field 'authors'
  warning 48: The fields 'build-test:' and 'build-doc:' are deprecated, and should be replaced by
              uses of the 'with-test' and 'with-doc' filter variables in the 'build:' and
              'install:' fields, and by the newer 'run-test:' field
ocameel is now pinned to file:///Users/ec/Dropbox/Code/ocameel (version 0.0.1)
```

**Comment by AltGr** [comment]
Thanks for reporting. Note that it's only a failed linting check: in other words, informative, but without any consequences on your package. It should be marked as `warning` rather than `error`, to be consistent, though.

**Comment by AltGr** [comment]
> is there any tooling or features for maintaining the .opam file? Do I need to edit this manually, while referring to the syntax documentation; or is there something along the lines of npm install --save, or npm init, to get opam to manage it for me?

Not at the moment, no. The only help is the `opam lint` command to check it for you.

**Comment by ELLIOTTCABLE** [comment]
Well, it’s more than a suggestion, at least until you change it to `warning` — because it causes the “automatic upgrade to v2.0 syntax” to fail, and leave the file as v1.2. Which means it re-attempts it every time I run an opam command. 🤣

**Comment by AltGr** [comment]
> “automatic upgrade to v2.0 syntax”

I'm not sure what you mean exactly ? Opam will do the upgrade internally, but not in your source (you can see the results with `opam show --raw PKG`, or `opam pin edit` ; or you can try [this](https://github.com/AltGr/opam-package-upgrade) tiny plugin to upgrade a single file)

**Comment by ELLIOTTCABLE** [comment]
Oh, wow — maybe I should have reported this as an unclear message, then?

Here's what it shows on my end:

```
[NOTE] Converting format of /Users/ec/Documents/Code/Source/ocaml-parsing/ocaml_parsing.opam from 1.2 to 2.0~beta5
```

That definitely read, to me, as “I'm automatically updating-in-place your configuration file to the new syntax.” Perhaps that needs a rephrase?

**Comment by AltGr** [comment]
Aah, I see! Indeed, sorry for the misleading message. Maybe add "internally", or even "Using compatibility layer for older format opam file %s".

Or we could actually skip the message entirely... I am not sure if that info is actually any use to the user.

**Comment by hcarty** [comment]
Skipping seems appropriate. A user consuming a package with 1.x metadata is unlikely to need the information. A user developing that package could get the information more directly, through their editor or maybe by `lint`.

**Comment by AltGr** [comment]
Addressed by 4303c76667e8cfe759861d868b403e93f39115f1. Thanks for reporting!
